tests/test_configuration.py

```diff
@@ -0,0 +1,87 @@
+import json
+import re
+from os import environ
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from lbz.aws_ssm import SSM
+from lbz.configuration import EnvValue, SSMValue
+from lbz.exceptions import ConfigValueParsingFailed, MissingConfigValue
+
+
+class TestConfigValue:
+    # We are using EnvValue instead ConfigValue in order to not create artificial class
+    # that wouldn't add any value.
+    @patch.dict(environ, {"key": "42"})
+    def test_if_getter_was_used_only_once(self) -> None:
+        cfg = EnvValue("key")
+
+        with patch.object(EnvValue, "getter", wraps=cfg.getter) as mocked_getter:
+            assert cfg.value == "42"
+            assert cfg.value == "42"
+
+        mocked_getter.assert_called_once()
+
+    def test_returns_default_when_no_value_set(self) -> None:
+        cfg = EnvValue("test", default=42, parser=int)
+
+        assert cfg.value == 42
+
+    def test_raises_missing_configuration_when_value_is_none(self) -> None:
+        cfg = EnvValue("test")
+
+        with pytest.raises(MissingConfigValue, match="'test' was not defined."):
+            cfg.value  # pylint: disable=pointless-statement
+
+    @patch.object(EnvValue, "getter", MagicMock(return_value=1))
+    def test_modifies_config_value_using_declared_parser_function(self) -> None:
+        cfg = EnvValue("RANDOM_VAR", lambda a: a * 10)
+
+        assert cfg.value == 10
+
+    @patch.object(EnvValue, "getter", MagicMock(return_value=(1, 2)))
```

Comment:
You should not mock getter, but rather adjust the test to show expected behavior based on the string value 😉